--- a/version.c	2024-12-25 16:43:20.000000000 +0900
+++ b/version.c	2026-04-04 21:26:14.018756287 +0900
@@ -148,9 +148,10 @@
 
 int ruby_mn_threads_enabled;
 
-#ifndef RB_DEFAULT_PARSER
-#define RB_DEFAULT_PARSER RB_DEFAULT_PARSER_PRISM
+#ifdef RB_DEFAULT_PARSER
+#undef RB_DEFAULT_PARSER
 #endif
+#define RB_DEFAULT_PARSER RB_DEFAULT_PARSER_PARSE_Y
 static ruby_default_parser_enum default_parser = RB_DEFAULT_PARSER;
 
 ruby_default_parser_enum
